src/store/auth.module.js

```diff
@@ -49,19 +49,25 @@ const actions = {
     })
   },
   [CHECK_AUTH] (context) {
-    if (JwtService.getToken()) {
-      ApiService.setHeader()
-      ApiService
-        .get('user')
-        .then(({data}) => {
-          context.commit(SET_AUTH, data.user)
-        })
-        .catch(({response}) => {
-          context.commit(SET_ERROR, response.data.errors)
-        })
-    } else {
-      context.commit(PURGE_AUTH)
-    }
+    return new Promise((resolve, reject) => {
+      if (JwtService.getToken()) {
+        ApiService.setHeader()
+        ApiService
+          .get('user')
+          .then(({data}) => {
+            context.commit(SET_AUTH, data.user)
+          })
+          .catch(({response}) => {
+            context.commit(SET_ERROR, response.data.errors)
+          })
+          .finally(() => {
+            resolve()
```

Comment:
Why not pass the user data to resolve within the 

```
.then
```
and
```
.catch
```
calls? 

This way we can use it with an auth library like [CASL](https://www.npmjs.com/package/@casl/vue).

Reply:
I think at the moment I wrote this, I want be sure that, this always be resolve so I use finally.
And about CASL I use vue-router meta for check if that page has required authentication or not, so I decide direct to login page or not. If you have time please check my little aside project: (I think this project not work anymore :D)
https://github.com/tranlehaiquan/conduit-vue